CMakeLists.txt

```diff
@@ -161,31 +165,48 @@ set(CMAKE_MESSAGE_LOG_LEVEL NOTICE)
 
 # Add third party subdirectories and define options appropriate to run their cmakes.
 set(pybind11_FIND_QUIETLY ON)
-add_subdirectory(third_party/onnx)
-add_subdirectory(third_party/pybind11)
-add_subdirectory(third_party/rapidcheck)
+
+if (ONNX_MLIR_USE_SYSTEM_ONNX)
+  find_package(ONNX 1.15.0 EXACT REQUIRED)
```

Comment:
I don't know why my comment disappeared so here it is again.

Instead of hardcoding 1.15.0, why not use
```
file(READ "${ONNX_MLIR_SRC_ROOT}/third_party/onnx/VERSION_NUMBER" ONNX_VERSION)
find_package(ONNX ${ONNX_VERSION} EXACT REQUIRED)
```